synapse/cortex.py

```diff
@@ -6391,6 +6393,76 @@ async def _setVaultSecrets(self, iden, obj, key, valu, delete):
         self.slab.put(bidn, s_msgpack.en(vault), db=self.vaultsdb)
         return data
 
+    async def replaceVaultConfigs(self, iden, valu):
+        '''
+        Replace the entire vault config.
+
+        Args:
+            iden (str): The iden of the vault to edit.
+            valu (str): New configs object to store on the vault.
+
+        Raises:
+            synapse.exc.BadArg: `valu` is not a dictionary.
+            synapse.exc.NoSuchIden: Vault with `iden` does not exist.
+            synapse.exc.NotMsgpackSafe: `valu` is not msgpack safe.
+
+        Returns: New configs.
+        '''
+        vault = self.reqVault(iden)
+
+        if not isinstance(valu, dict):
+            raise s_exc.BadArg(mesg='valu must be a dictionary.', name='valu', valu=valu)
+
+        try:
+            s_msgpack.en(valu)
+        except s_exc.NotMsgpackSafe:
+            raise s_exc.NotMsgpackSafe(
+                mesg='Vault configs must be msgpack safe.',
+                name='valu',
+                valu=valu) from None
```

Comment:
Looking back over these errors. We need to `repr()` the `valu` object and cap it at some reasonable length ( ex: `valu=repr(valu)[:256]` ). Otherwise when we try to push this exception over to a caller, they won't be able to receive the msgpack'd `valu`.